packages/main/src/DayPicker.ts

```diff
@@ -290,8 +289,8 @@ class DayPicker extends CalendarPart implements ICalendarPicker {
 				day.classes += " ui5-dp-item--othermonth";
 			}
 
-			if (isWeekend) {
-				day.classes += " ui5-dp-item--weeekend";
+			if ((isWeekend || specialDayType === "NonWorking") && specialDayType !== "Working") {
```

Comment:
We are only searching for a single special date object related to a given date.
It is possible by design to have more than one special date corresponding to a date. That way it will be possible to
display a `NonWorking` day and a `Type10` for example at the same time.
Additionally if there are `Working` and `NonWorking` types for a date then the first one will take precedence.
We'll have to use `filter` at line 230 instead of `find`.